OrbitQt/TopDownViewItemModel.cpp

```diff
@@ -0,0 +1,209 @@
+// Copyright (c) 2020 The Orbit Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+#include "TopDownViewItemModel.h"
+
+TopDownViewItemModel::TopDownViewItemModel(
+    std::unique_ptr<TopDownView> top_down_view, QObject* parent)
+    : QAbstractItemModel{parent}, top_down_view_{std::move(top_down_view)} {}
+
+QVariant TopDownViewItemModel::data(const QModelIndex& index, int role) const {
+  if (!index.isValid()) {
+    return QVariant();
+  }
+  switch (role) {
+    case Qt::DisplayRole: {
+      auto* item = static_cast<TopDownNode*>(index.internalPointer());
+      auto thread_item = dynamic_cast<TopDownThread*>(item);
+      auto function_item = dynamic_cast<TopDownFunction*>(item);
+      if (thread_item != nullptr) {
```

Comment:
Indentation is getting pretty deep here. Can you refactor this to move some parts out into functions?

Reply:
Thanks, looks good.